zellij-server/src/tab/mod.rs

```diff
@@ -3280,172 +3297,6 @@ impl Tab {
-                );
-
-                relative_position.change_line(
-                    (relative_position.line())
-                        .max(0)
-                        .min(active_pane.get_content_rows() as isize),
-                );
-                if let Some(mouse_event) = active_pane.mouse_right_click(&relative_position, true) {
-                    self.write_to_active_terminal(
-                        &None,
-                        mouse_event.into_bytes(),
-                        false,
-                        client_id,
-                    )
-                    .with_context(err_context)?;
-                    return Ok(true); // we need to re-render in this case so the selection disappears
-                }
-            }
-        }
-        Ok(false) // we shouldn't even get here, but might as well not needlessly render if we do
-    }
-
-    pub fn handle_mouse_hold_middle(
-        &mut self,
-        position_on_screen: &Position,
-        client_id: ClientId,
-    ) -> Result<bool> {
-        let err_context = || {
-            format!("failed to handle left mouse hold at position {position_on_screen:?} for client {client_id}")
-        };
-        // return value indicates whether we should trigger a render
-        // determine if event is repeated to enable smooth scrolling
-        let is_repeated = if let Some(last_position) = self.last_mouse_hold_position {
-            position_on_screen == &last_position
-        } else {
-            false
-        };
-        self.last_mouse_hold_position = Some(*position_on_screen);
-
-        let active_pane = self.get_active_pane_or_floating_pane_mut(client_id);
-
-        if let Some(active_pane) = active_pane {
-            let mut relative_position = active_pane.relative_position(position_on_screen);
-            if !is_repeated {
-                relative_position.change_column(
-                    (relative_position.column())
-                        .max(0)
-                        .min(active_pane.get_content_columns()),
-                );
-
-                relative_position.change_line(
-                    (relative_position.line())
-                        .max(0)
-                        .min(active_pane.get_content_rows() as isize),
-                );
-                if let Some(mouse_event) = active_pane.mouse_middle_click(&relative_position, true)
-                {
-                    self.write_to_active_terminal(
-                        &None,
-                        mouse_event.into_bytes(),
-                        false,
-                        client_id,
-                    )
-                    .with_context(err_context)?;
-                    return Ok(true); // we need to re-render in this case so the selection disappears
-                }
-            }
-        }
-        Ok(false) // we shouldn't even get here, but might as well not needlessly render if we do
-    }
-
```

Comment:
The three functions above are decently large.  Is there logic anywhere else to handle changing the selection buffer size as the mouse is dragged around?  If not, then we will need to put that in handle_mouse_event().